solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/test/java/com/solace/spring/cloud/stream/binder/util/XMLMessageMapperTest.java

```diff
@@ -416,6 +418,55 @@ public void testMapConsumerSpringMessageToXMLMessage_WithProperties() {
 		assertEquals(consumerProperties.getRepublishedMsgTtl().longValue(), xmlMessage.getTimeToLive());
 	}
 
+	@Test
+	public void testMapProducerSpringMessageToXMLMessage_WithExcludedHeader() throws SDTException {
+		String testPayload = "testPayload";
+		List<String> excludedHeaders = new ArrayList<>(Arrays
+				.asList("io.opentracing.contrib.spring.integration.messaging.OpenTracingChannelInterceptor.SCOPE"));
+		Message<?> testSpringMessage = new DefaultMessageBuilderFactory().withPayload(testPayload)
+				.setHeader(
+						"io.opentracing.contrib.spring.integration.messaging.OpenTracingChannelInterceptor.SCOPE",
+						"any")
+				.build();
+
+		XMLMessage xmlMessage = xmlMessageMapper.map(testSpringMessage, excludedHeaders);
+		Mockito.verify(xmlMessageMapper).map(testSpringMessage, excludedHeaders);
+
+		assertEquals(null, xmlMessage.getProperties()
+				.getMap("io.opentracing.contrib.spring.integration.messaging.OpenTracingChannelInterceptor.SCOPE"));
+	}
+
+	@Test
+	public void testMapProducerSpringMessageToXMLMessage_WithExcludedHeader_ShouldNotMatchPartially() throws SDTException {
+		String testPayload = "testPayload";
+		List<String> excludedHeaders = new ArrayList<>(Arrays
+				.asList("io.opentracing.contrib.spring.integration.messaging.OpenTracingChannelInterceptor.SCOPE"));
+		Message<?> testSpringMessage = new DefaultMessageBuilderFactory().withPayload(testPayload)
+				.setHeader(
+						"io.opentracing.contrib.spring.integration.messaging.OpenTracingChannelInterceptor",
+						"any")
+				.build();
+
+		XMLMessage xmlMessage = xmlMessageMapper.map(testSpringMessage, excludedHeaders);
+		Mockito.verify(xmlMessageMapper).map(testSpringMessage, excludedHeaders);
+		assertEquals("any", xmlMessage.getProperties()
+				.get("io.opentracing.contrib.spring.integration.messaging.OpenTracingChannelInterceptor"));
+	}
+
+	@Test
+	public void testMapProducerSpringMessageToXMLMessage_WithExcludedHeader_ShouldNotFilterSolaceHeader() throws SDTException {
```

Comment:
Is this test enough? Or do we have to check all solace headers are not filtered?

Reply:
This is good enough for the headers defined in `SolaceHeaders` since those are set generically set on the resulting `XMLMessage`.

But for each header defined in `SolaceBinderHeaders` which has scope `WIRE` (as per `SolaceBinderHeaderMeta`), could you validate that they cannot be filtered?

Its sufficient just to check that each of the `SolaceBinderHeaders` do not have a `null` value in the resulting `XMLMessage`'s property map. And to the populate the serialization headers, you can use the `SerializableFoo` class as the value for a Spring header.

Reply:
agree, test no checks all header